inc/platform/vt_platform.h

```diff
@@ -8,21 +8,31 @@
 
 #include "vt_defs.h"
 
-typedef VT_UINT (*VT_ADC_FUNC_INIT)(VT_ADC_ID adc_id,
+typedef VT_UINT (*VT_ADC_SINGLE_READ_INIT_FUNC)(VT_ADC_ID adc_id,
     VT_ADC_CONTROLLER* adc_controller,
     VT_ADC_CHANNEL* adc_channel,
     VT_UINT* adc_resolution,
     VT_FLOAT* adc_ref_volt);
-typedef VT_UINT (*VT_ADC_FUNC)(VT_ADC_ID adc_id, VT_ADC_CONTROLLER* adc_controller, VT_ADC_CHANNEL* adc_channel);
+typedef VT_UINT (*VT_ADC_SINGLE_READ_FUNC)(VT_ADC_ID adc_id, VT_ADC_CONTROLLER* adc_controller, VT_ADC_CHANNEL* adc_channel);
+typedef VT_VOID (*VT_ADC_BUFFER_READ_CALLBACK_FUNC)(void);
+typedef VT_UINT (*VT_ADC_BUFFER_READ_FUNC)(VT_ADC_ID adc_id,
+    VT_ADC_CONTROLLER* adc_controller,
+    VT_ADC_CHANNEL* adc_channel,
+    VT_FLOAT* adc_read_buffer,
+    VT_UINT buffer_length,
+    VT_FLOAT sampling_frequency,
+    VT_ADC_BUFFER_READ_CALLBACK_FUNC vt_adc_buffer_read_conv_half_cplt_callback,
+    VT_ADC_BUFFER_READ_CALLBACK_FUNC vt_adc_buffer_read_conv_cplt_callback);
 typedef VT_UINT (*VT_GPIO_FUNC)(VT_GPIO_ID gpio_id, VT_GPIO_PORT* gpio_port, VT_GPIO_PIN* gpio_pin);
 typedef VT_UINT (*VT_TICK_INIT_FUNC)(VT_UINT* max_value, VT_UINT* resolution_usec);
 typedef VT_ULONG (*VT_TICK_FUNC)();
 typedef VT_VOID (*VT_INTERRUPT_CTRL)();
 
 typedef struct VT_DEVICE_DRIVER_STRUCT
 {
-    VT_ADC_FUNC_INIT adc_init;
-    VT_ADC_FUNC adc_read;
+    VT_ADC_SINGLE_READ_INIT_FUNC adc_single_read_init;
+    VT_ADC_SINGLE_READ_FUNC adc_single_read;
+    VT_ADC_BUFFER_READ_FUNC adc_buffer_read;
```

Comment:
is there a buffer_read_init required?

Reply:
It is not required, but can add for symmetry

Reply:
I dont think it needs to be added for symmetry, but how comes one isnt needed? Is init covered in the adc_single_read_init function?